### DISCUSSION TITLE
host hugo site on github

### DISCUSSION THREAD

**Discussion by varghesejose2020** [EddieHubCommunity/support#1365] [Q&A]
hi
I created a hugo themed site(i used compose theme) i want to host my site on github
but i tried it doesnot work
any idea how host  in github

**Comment by kaiwalyakoparkar** [comment]
could you please share the link to the repo

**Comment by schmelto** [comment]
May take a look at this link: https://gohugo.io/hosting-and-deployment/hosting-on-github/

**Reply by varghesejose2020** [reply]
![err](https://user-images.githubusercontent.com/71702982/115566168-a41d8b00-a2d7-11eb-8feb-62511456c9bd.JPG)

i run git init shows

**Reply by schmelto** [reply]
Thats good. Did you push your changes/repository to GitHub?
After publish the repo to GitHub add the GitHub action described in the link above.

**Comment by varghesejose2020** [comment]
![err1](https://user-images.githubusercontent.com/71702982/115567258-b21fdb80-a2d8-11eb-9f09-d417d541bc68.JPG)
how to resolve thiese

**Reply by schmelto** [reply]
Git add "filename"

**Reply by schmelto** [reply]
I see you're using VSCode you can also initialize Github (https://vscode.github.com/)

**Comment by varghesejose2020** [comment]
![err2](https://user-images.githubusercontent.com/71702982/115568644-20b16900-a2da-11eb-9bd3-1ddb306f6ce6.JPG)

**Reply by schmelto** [reply]
try a `git push` [see here](https://git-scm.com/docs/git-push) think with the git init at the beginning all files are currently in the commit.

This link is also great
https://www.datacamp.com/community/tutorials/git-push-pull

**Comment by varghesejose2020** [comment]
![err35](https://user-images.githubusercontent.com/71702982/115662037-028c4d00-a35c-11eb-8c1a-d7011a4cfe8d.JPG)

**Comment by schmelto** [comment]
`git init`

`git remote add origin <yorgitrepository.git>`

`git commit -m 'test.txt'` etc.

`git push -u origin master` or `main` whatever you prefer

you can check with `git status` if there are any untracked files in the repo

![image](https://user-images.githubusercontent.com/30869493/115664986-e92ad700-a342-11eb-9330-61cb5748ea53.png)

if the files are on github you can go into the `.github/workflows`-folder and create a file `gh-pages.yml` with following code: https://gohugo.io/hosting-and-deployment/hosting-on-github/#build-hugo-with-github-action

**Reply by schmelto** [reply]
it might be that you have not to commit anything due the git init I'm not completely sure

**Comment by varghesejose2020** [comment]
![err56](https://user-images.githubusercontent.com/71702982/115665841-72e99d00-a361-11eb-95b0-8bde27aba7fb.JPG)
my login credentials are correct but it shows
